src/coreclr/src/vm/stubmgr.cpp

```diff
@@ -1786,41 +1786,41 @@ BOOL ILStubManager::TraceManager(Thread *thread,
         LOG((LF_CORDB, LL_INFO10000, "ILSM::TraceManager: Unmanaged CALLI case 0x%x\n", target));
         trace->InitForUnmanaged(target);
     }
-    else if (pStubMD->IsStructMarshalStub())
+    else if (pStubMD->IsPInvokeStub())
```

Comment:
Does the implementation of `IsPInvokeStub` handle other stub types like instantiating stubs correctly? It looks like its definition might not fully cover those cases.

Reply:
You are actually right that some things fall through these/don't get correctly classified. I'll need to dive deeper to see why this is happening